packages/storage-plus/src/prefix.rs

```diff
@@ -65,31 +66,43 @@ impl<'a, K: Prefixer<'a>> PrefixBound<'a, K> {
     }
 }
 
-type DeserializeFn<T> = fn(&dyn Storage, &[u8], Pair) -> StdResult<Pair<T>>;
+type DeserializeKvFn<K, T> =
+    fn(&dyn Storage, &[u8], Pair) -> StdResult<(<K as KeyDeserialize>::Output, T)>;
 
-pub fn default_deserializer<T: DeserializeOwned>(
+#[allow(dead_code)]
+pub fn default_deserializer_v<T: DeserializeOwned>(
     _: &dyn Storage,
     _: &[u8],
     raw: Pair,
 ) -> StdResult<Pair<T>> {
-    deserialize_kv(raw)
+    deserialize_v(raw)
+}
+
+pub fn default_deserializer_kv<K: KeyDeserialize, T: DeserializeOwned>(
+    _: &dyn Storage,
+    _: &[u8],
+    raw: Pair,
+) -> StdResult<(K::Output, T)> {
+    deserialize_kv::<K, T>(raw)
 }
 
 #[derive(Clone)]
-pub struct Prefix<T>
+pub struct Prefix<K = Vec<u8>, T = Vec<u8>>
```

Comment:
I don't see any need for the defaults. We didn't have one before/
T will never be default and if we specify T, we cannot use the K default.

Just make it <K, T>

Reply:
Good point. Will remove.